app/domain/etl/item/processor.rb

```diff
@@ -1,25 +1,46 @@
 class Etl::Item::Processor
-  attr_reader :item, :date
+  attr_reader :new_item, :old_item, :dimensions_date
 
-  def self.run(item, date)
-    new(item: item, date: date).run
+  def self.run(new_item, old_item, date)
+    new(new_item: new_item, old_item: old_item, date: date).run
   end
 
-  def initialize(item:, date:)
-    @item = item
-    @date = date
+  def initialize(new_item:, old_item:, date:)
+    @new_item = new_item
+    @old_item = old_item
+    @dimensions_date = Dimensions::Date.for(date)
   end
 
   def run
-    edition = Facts::Edition.create!(
-      number_of_pdfs: Etl::Item::Metadata::NumberOfPdfs.parse(item.raw_json),
-      number_of_word_files: Etl::Item::Metadata::NumberOfWordFiles.parse(item.raw_json),
-      dimensions_date: Dimensions::Date.for(date),
-      dimensions_item: item,
-    )
-
-    if item.document_text.present?
-      edition.update(Etl::Item::Quality::Service.new.run(item.document_text))
+    if content_changed?
+      create_new_edition
+    else
+      clone_existing_edition
     end
   end
+
+  def content_changed?
+    return true unless old_item
```

Comment:
Could you please check what happens with both content items have no text (return `nil`)? This method will return `true` but in reality should be `false`?

Reply:
It does return the correct thing.I just added an extra test to cover this scenario.